### PR TITLE
Remove Electrum support

### PR DESCRIPTION
**Pull request by t-bast**
Electrum support was provided for mobile wallets, server nodes should always run a bitcoind node as this provides more control (especially for utxo management for anchor outputs channels).

Since wallets will use https://github.com/acinq/eclair-kmp instead of eclair, we can now remove Electrum and API fee providers from eclair.

10k lines of code removed, it's quite a feat! Eclair is now down to ~30k lines of code ;)

### CODE REVIEW

File: eclair-core/src/main/scala/fr/acinq/eclair/blockchain/WatcherTypes.scala [hunk 1]
```diff
@@ -43,7 +43,7 @@ sealed trait Watch {
  *
  * @param replyTo         actor to notify once the transaction is confirmed.
  * @param txId            txid of the transaction to watch.
- * @param publicKeyScript when using electrum, we need to specify a public key script; any of the output scripts should work.
+ * @param publicKeyScript output script of any of the transaction's outputs.
```
Comment (t-bast) [inline eclair-core/src/main/scala/fr/acinq/eclair/blockchain/WatcherTypes.scala:46]: Should I completely remove these fields? They were only necessary for Electrum.

File: eclair-core/src/main/resources/reference.conf [hunk 2]
```diff
@@ -20,7 +20,7 @@ eclair {
   // override this with a script/exe that will be called everytime a new database backup has been created
   # backup-notify-script = "/absolute/path/to/script.sh"
 
-  watcher-type = "bitcoind" // other *experimental* values include "electrum"
+  watcher-type = "bitcoind"
```
Comment (pm47) [inline eclair-core/src/main/resources/reference.conf:23]: We could remove this. Will we add a new watcher type in a near future?

File: eclair-core/src/main/scala/fr/acinq/eclair/blockchain/WatcherTypes.scala (hunk 1 above)
Comment (pm47) [reply eclair-core/src/main/scala/fr/acinq/eclair/blockchain/WatcherTypes.scala:46]: I think yes.

Comment (t-bast) [reply eclair-core/src/main/scala/fr/acinq/eclair/blockchain/WatcherTypes.scala]: Done in https://github.com/ACINQ/eclair/pull/1750/commits/447cba4187a56c7d974c73feb3e533d8f7369708

File: eclair-core/src/main/resources/reference.conf (hunk 2 above)
Comment (t-bast) [reply eclair-core/src/main/resources/reference.conf:23]: Agreed, we can remove it. I don't think we'll support a new watcher type in the near future.

Comment (t-bast) [reply eclair-core/src/main/resources/reference.conf]: Done in https://github.com/ACINQ/eclair/pull/1750/commits/9771ce6d4868ac7955f6d338dc9f70779d523f88

**Review by pm47 — APPROVED** [review]

**Review by pm47 — APPROVED** [review]

**Pull request merged**